app/models/solidus_bolt/gateway.rb

```diff
@@ -51,8 +51,22 @@ def credit(_amount, _response_code, _gateway_options)
       raise NotImplementedError, 'credit method has not been implemented in SolidusBolt::Gateway class'
     end
 
-    def purchase(_float_amount, _response_code, _gateway_options)
-      raise NotImplementedError, 'purchase method has not been implemented in SolidusBolt::Gateway class'
+    def purchase(_float_amount, payment_source, gateway_options)
+      order_number = gateway_options[:order_id].split('-').first
+      order = Spree::Order.find_by(number: order_number)
+
+      authorization_response = ::SolidusBolt::Transactions::AuthorizeService.call(
+        order: order,
+        create_bolt_account: payment_source.create_bolt_account,
+        credit_card: credit_card_params(payment_source),
+        payment_method: payment_source.payment_method,
+        auto_capture: true
+      )
```

Comment:
What if instead of passing `auto_capture: true` we first call AuthorizeService and then CaptureService?

When implementing the AuthorizeService we talked about the auto_capture option and was decided to leave it as always false because we should handle capture via the inbuilt Solidus' flow

Thoughts @DanielePalombo ?

Reply:
The in-built solidus flow offers 2 options.

1) auto_capture == true -> gateway#purchase
2) auto_capture == false -> gateway#authorize && gateway#capture

In my opinion the change made by Piyush is valid.